---
title: "K-means clustering"
author: "Allison Tai"
date: "March 15, 2017"
output: 
  html_document: 
    keep_md: yes
---
# source("http://bioconductor.org/biocLite.R") 
# biocLite(c("AnnotationDbi", "impute", "GO.db", "preprocessCore"))
# install.packages(WGCNA)

```{r, message=FALSE, warning=FALSE}
source("https://bioconductor.org/biocLite.R")
biocLite('biomaRt')
library(tidyverse)
library(magrittr)
library(scatterplot3d)
library(biomaRt)
options(stringsAsFactors = FALSE)
```

```{r}
getwd()
countdata <- read.table(file= "../Raw_Data/GSE85567_RNASeq_normalizedcounts.txt", check.names = FALSE)
metadata <- read.csv(file= "../Raw_Data/GSE85566_metadata.txt", row.names = 1)

metadata %<>% filter(ID %in% colnames(countdata))
metadata %>% group_by(Status) %>% tally()

colnames(countdata) == metadata$ID
countdata <- countdata[,metadata$ID]
colnames(countdata) == metadata$ID
```

Extract the three relevant genes
```{r}
mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
# time to grab the gene names to find our three genes
countdata
genes <- rownames(countdata)
geneList <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id",
"entrezgene", "description"),values=genes,mart= mart)
# from our genes list, it looks like:
# CLCA1 = ENSG00000016490
# serpinB2 = ENSG00000197632
# periostin = ENSG00000133110

# now grab those rows
trundata <- countdata[c("ENSG00000016490", "ENSG00000197632", "ENSG00000133110"),]
```

Let's cluster
```{r}
# check to make sure data is okay
tcount <- as.data.frame(t(countdata))
gsg <- goodSamplesGenes(tcount, verbose = 1)
gsg$allOK # All are good. If not, will be False and we would have to filter them

# now work with our genes
ttrun <- as.data.frame(t(trundata))
ttrun <- log2(ttrun)

# replace -inf with 0
ttrun[ttrun == -Inf] <- 0
# center and scale our gene expression
ttrunSC <- data.frame(scale(ttrun))

# separate asthmatics and control into two matrices using our added column
matrixList <- split(ttrunSC, metadata$Status)

# now grab the patients and cluster
asthma <- data.frame(matrixList$Asthma)
control <- data.frame(matrixList$Control)

geneCluster = kmeans(asthma,2,100,nstart=20);

geneCluster$cluster <- as.factor(geneCluster$cluster)
sizeGrWindow(12,12)

# add clusters to metadata
metadata$cluster <- 0
for (i in 1:57) {
 x <- which(metadata$ID == rownames(asthma)[i])
  metadata[x, 9] = geneCluster$cluster[i]
}
write.csv(metadata, "metaCluster.csv")

# save everything!
write.table(asthma, file = "asthma.tsv")
write.table(control, file = "control.tsv")
```

Now let's draw some graphs and look at our clusters
```{r, eval=FALSE, include=FALSE}
library(ggplot2)
library(gridExtra)
library(scatterplot3d)
# we're not going to re-generate our k-means clusters anymore, so run starting from here:
metadata <- read.csv('metaCluster.csv', row.names = 1)
asthma <- read.table('asthma.tsv')
control <- read.table('control.tsv')

# CLCA1 = ENSG00000016490
# serpinB2 = ENSG00000197632
# periostin = ENSG00000133110

names(asthma)[names(asthma)=="ENSG00000016490"] <- "CLCA1"
names(asthma)[names(asthma)=="ENSG00000197632"] <- "SerpinB2"
names(asthma)[names(asthma)=="ENSG00000133110"] <- "Periostin"

# re-make the clusters, because I killed them
geneCluster <- data.frame(cluster = rep("", nrow(asthma)), stringsAsFactors = FALSE)
rownames(geneCluster) <- rownames(asthma)
for (i in 1:57) {
  x <- which(metadata$ID == rownames(geneCluster)[i])
  if (metadata[x,9] == 1)
    geneCluster[i,1] = "Th2-high"
  else
    geneCluster[i,1] = "Th2-low"
}

# let's make some plots
p1 <- ggplot(asthma, aes(CLCA1, SerpinB2, color = geneCluster$cluster)) + geom_point() + scale_colour_manual(values=c("#34675C", "#7DA3A1"), name = "Cluster")
p2 <- ggplot(asthma, aes(CLCA1, Periostin, color = geneCluster$cluster)) + geom_point() + scale_colour_manual(values=c("#34675C", "#7DA3A1"), name = "Cluster")
p3 <- ggplot(asthma, aes(SerpinB2, Periostin, color = geneCluster$cluster)) + geom_point() + scale_colour_manual(values=c("#34675C", "#7DA3A1"), name = "Cluster")
grid.arrange(p1, p2, p3, nrow = 3)

# save
g <- arrangeGrob(p1, p2, p3, nrow=3) #generates g
ggsave(file="k_means.png", g) #saves g
```

Now let's look at our clusters with a 3D scatterplot.
```{r, eval=FALSE, include=FALSE}
png("3d.png", width = 7, height = 7, units = 'in', res = 300)
attach(asthma)
# define colours for the clusters
geneCluster$color[geneCluster$cluster == "Th2-high"] <- "#34675C"
geneCluster$color[geneCluster$cluster == "Th2-low"] <- "#7DA3A1"
# plot
scatterplot3d(CLCA1, SerpinB2, Periostin, color = geneCluster$color, pch=19, ylab = "")
legend("topleft", inset=.05,      # location and inset
    bty="n", cex=.7,              # suppress legend box, shrink text 50%
    title="Cluster",
    c("Th2-high", "Th2-low"), fill=c("#34675C", "#7DA3A1"))
text(x = 5, y = -1.8, "SerpinB2", srt = 45)
detach(asthma)
# save plot
dev.off()
```

Now let's re-cluster with our controls and look at the scatterplots.
```{r, eval=FALSE, include=FALSE}
fused <- rbind(asthma, control)
gcFused = kmeans(fused,3,100,nstart=20);

#attach(fused)
#scatterplot3d(ENSG00000016490, ENSG00000197632, ENSG00000133110, color = gcFused$cluster)

q1 <- ggplot(fused, aes(ENSG00000016490, ENSG00000197632, color = gcFused$cluster)) + geom_point()
q2 <- ggplot(fused, aes(ENSG00000016490, ENSG00000133110, color = gcFused$cluster)) + geom_point()
q3 <- ggplot(fused, aes(ENSG00000133110, ENSG00000197632, color = gcFused$cluster)) + geom_point()
grid.arrange(q1, q2, q3, nrow = 3)

# create a new metadata, just so we can look at it
mdTrial <- metadata
for (i in 1:85) {
 x <- which(mdTrial$ID == rownames(fused)[i])
  mdTrial[x, 9] = gcFused$cluster[i]
}

```

Looks like 2 is lowest, 3 is medium, and 1 is high.

Extract the pathways associated with CLCA1, periostin, and serpinB2

```{r}
#library('sigora')
# 5055 is serpinB2
# 1179 is CLCA1
# 10631 is periostin
#pathways <- ora(c(1179), kegH)
``````
