---
title: "Differential Expression 3 Group"
author: "Arjun Baghela"
date: "3/29/2017"
output: github_document
---
Our last step was [k-means clustering](https://github.com/STAT540-UBC/team_Undecided/blob/master/src/2_kmeans_clustering/Cluster.Rmd), where we assigned each asthma patient to a group (Th2-high or Th2-low) depending on their expression in three genes.  

In this document, we will perform differential expression analysis of the RNA-seq data between the three groups (control, high, and low).  The resulting three lists (control vs high, control vs low, and high vs low) will be combined, and we'll filter out about 500 genes (deemed "interesting"), which will be passed to the next stage for constructing the correlation network and performing differential expression analysis.  

Load the necessary packages. 
```{r, results='asis', warning=FALSE}
library(edgeR)
library(limma)
library(tidyverse)
library(magrittr)
```

Load in the normalized RNA-seq count data.
```{r}
countdata <- read.table(file= "../../data/raw_data/rna_seq_data/GSE85567_RNASeq_normalizedcounts.txt", check.names = FALSE)
metadata <- read.csv(file= "../../data/processed_data/metaCluster.csv", row.names = 1)

metadata %<>% filter(ID %in% colnames(countdata)) # Remove metadata rows that do not have

metadata %>% group_by(Status) %>% tally() # See how many patients there are in each group

metadata %<>% arrange(cluster) # arrange by classes
metadata %>% group_by(cluster) %>% tally() # how many patients are there in each cluster- Control, Th2 high (1), Th2 low (2). 

countdata <- countdata[,as.character(metadata$ID)] # Maintain the same order in both metadata and count data
colnames(countdata) == metadata$ID

metadata$cluster <- as.factor(metadata$cluster)
str(metadata)

remove <- c("ENSG00000016490", "ENSG00000197632", "ENSG00000133110") # Remove the three genes we clustered on, to remove any biases. 
countdata <- countdata[!(rownames(countdata) %in% remove),]
```

We'll process the data some more.
```{r}
DGElist <- DGEList(counts= countdata, group= metadata$cluster) # create DGEList to store data in
DGElist$samples$lib.size %>% min() # Find lib size
DGElist$counts %>% nrow()

keep <- rowSums(cpm(DGElist)>0.275) >= 28 # Filter lowly expressed genes. Genes with less than 5 counts as determined by library with smallest size. 

DGElistFilt <- DGElist[keep, , keep.lib.sizes=FALSE] 
DGElistFilt$counts %>% nrow()

DGElistFiltNorm<- calcNormFactors(DGElistFilt) # calculate Norm factors
```

Now, let's perform the differential expression analysis, using edgeR.
```{r}
design <- model.matrix(~0+group, data=DGElistFiltNorm$samples) #Create model matrix
design %>% head()

DGElistFiltNormDisp <- estimateDisp(DGElistFiltNorm, design) # Calculate dispersion 
fit <- glmFit(DGElistFiltNormDisp, design) #GLM fit

lrt <- glmLRT(fit, contrast=c(-1,1,0)) 
filtHighCon <- (topTags(lrt, n= Inf))$table %>% rownames_to_column(var="Gene") %>% filter(FDR <= .01)

lrt <- glmLRT(fit, contrast=c(-1,0,1))
filtLowCon <- (topTags(lrt, n= Inf))$table %>% rownames_to_column(var="Gene") %>% filter(FDR <= .25)

lrt <- glmLRT(fit, contrast=c(0,1,-1))
filtHighLow <- (topTags(lrt, n= Inf))$table %>% rownames_to_column(var="Gene") %>% filter(FDR <= .01)

union <- union(filtHighCon$Gene, filtLowCon$Gene) %>% union(filtHighLow$Gene) # Union of Important genes. 
union %>% length() # List is 571 genes to do differential co-expression analysis on now. 

write.table(union, "../../data/processed_data/DEGene_ForDiffCoexpAnalysis.txt") # Write it to a table. 
```
Now, we can move onto the next stage, [differential methylation](https://github.com/STAT540-UBC/team_Undecided/blob/master/src/1_data_inspect_%26_4_diff%20met/Cleaning_methylation_data.Rmd#assessment-of-differentially-methylated-sites), which is necessary for weighting the correlation network we'll generate later.  



##### DONT LOOK AT THIS. 

PRODUCES THE SAME RESULTS AS ABOVE. 
```{r, eval=FALSE}
# model.matrix(~cluster*Gender*Age*current_smoker, metadata) %>% colnames()

design <- model.matrix(~group, data=DGElistFiltNorm$samples)

colnames(design) <- c("(Intercept)", "Th2 High", "Th2 Low")

DGElistFiltNormDisp <- estimateDisp(DGElistFiltNorm, design)
plotBCV(DGElistFiltNormDisp)
```

```{r, eval=FALSE}
fit <- glmFit(DGElistFiltNormDisp, design)
lrtHigh_Con <- glmLRT(fit, coef = 2)
lrtLow_Con <- glmLRT(fit, coef = 3)

DEHigh_Con <- (topTags(lrtHigh_Con, n= Inf))$table
DELow_Con <- (topTags(lrtLow_Con, n= Inf))$table

DEHigh_ConFilt <- DEHigh_Con %>% rownames_to_column(var="Gene") %>% filter(FDR <= .25)
DELow_ConFilt <- DELow_Con %>% rownames_to_column(var="Gene") %>% filter(FDR <= .25)

DEHigh_ConFilt %>% nrow()
DELow_ConFilt %>% nrow()

intersect(DEHigh_ConFilt$Gene,DELow_ConFilt$Gene) %>% length()
finalGeneList <- union(DEHigh_ConFilt$Gene,DELow_ConFilt$Gene)
```

```{r, eval= FALSE}
DGElistFiltNorm2 <- DGElistFiltNorm
DGElistFiltNorm2$samples$group <- relevel(DGElistFiltNorm$samples$group, ref="2")

design2 <- model.matrix(~group, data=DGElistFiltNorm2$samples)

DGElistFiltNorm2Disp <- estimateDisp(DGElistFiltNorm2, design2)

fit2 <- glmFit(DGElistFiltNorm2Disp, design2)
lrt2 <-  glmLRT(fit2, coef=3)

DEHigh_Low <- (topTags(lrt2, n= Inf))$table

DEHigh_LowFilt <- DEHigh_Low %>% rownames_to_column(var="Gene") %>% filter(FDR <= .25)
DEHigh_LowFilt %>% nrow()

intersect(finalGeneList, DEHigh_LowFilt$Gene) %>% length()
```





